src/linker/Linker/RemoveAttributeInstancesAttribute.cs

```diff
@@ -14,14 +16,14 @@ namespace Mono.Linker
 		AttributeTargets.Class, Inherited = false)]
 	public sealed class RemoveAttributeInstancesAttribute : Attribute
 	{
-		public RemoveAttributeInstancesAttribute ()
+		public RemoveAttributeInstancesAttribute (System.Collections.Generic.ICollection<CustomAttributeArgument> values)
 		{
-			Arguments = Array.Empty<CustomAttributeArgument> ();
-		}
-
-		public RemoveAttributeInstancesAttribute (CustomAttributeArgument value1)
-		{
-			Arguments = new[] { value1 };
+			// Arguments must be boxed in an Object CustomArgumentAttribute where the .Value is a CustomArgumentAttribute
+			Debug.Assert(values.All (a => a.Value is CustomAttributeArgument));
+			Arguments = values
```

Comment:
Why is this needed if we control what is passed in?

Reply:
Since the list of arguments is filtered to only the boxed values to avoid crashing, I put a sanity check in there to avoid a situation where a new call to the ctor mistakenly passes in a list of "unboxed" arguments and later finds that none of them are actually stored in the property.

Reply:
I think it'd be better to show error message when someone enters wrong data instead of skipping wrong entries and possibly matching different overload

Reply:
It looks like elsewhere we do ensure that the argument will be boxed in an `Object`. I'll leave the `Debug.Assert`, but I'll remove the `.Where` to make sure we don't silently match the wrong constructor.